pkg/kubectl/rolling_updater.go

```diff
@@ -531,21 +531,38 @@ func LoadExistingNextReplicationController(c client.ReplicationControllersNamesp
 	return newRc, err
 }
 
-func CreateNewControllerFromCurrentController(c *client.Client, namespace, oldName, newName, image, deploymentKey string) (*api.ReplicationController, error) {
+func CreateNewControllerFromCurrentController(c *client.Client, namespace, oldName, newName, image, container, deploymentKey string) (*api.ReplicationController, error) {
+	containerIndex := 0
 	// load the old RC into the "new" RC
 	newRc, err := c.ReplicationControllers(namespace).Get(oldName)
 	if err != nil {
 		return nil, err
 	}
 
-	if len(newRc.Spec.Template.Spec.Containers) > 1 {
-		// TODO: support multi-container image update.
-		return nil, goerrors.New("Image update is not supported for multi-container pods")
+	if container != "" {
```

Comment:
We follow a convention of comparing the length of a string against zero instead of an empty string. So this should be `len(container) != 0`.